lua/barbar/groups.lua

```diff
@@ -0,0 +1,170 @@
+--
+-- groups.lua
+--
```

Comment:
I'm open to finding a better name for these.

Reply:
My suggestions would be:

| `group_clump` | `group` | reference |
|:--|:--|:--|
| `division` | `subdivision` | |
| `element` | `content` | HTML |
| `group` | `atom` | regex |
| `segment` | `component` | other statusline plugins |